engine/runtime/src/main/java/org/enso/interpreter/node/callable/dispatch/LoopingCallOptimiserNode.java

```diff
@@ -79,13 +82,22 @@ public Object uncachedDispatch(
       State state,
       Object[] arguments,
       @Cached ExecuteCallNode executeCallNode) {
+    Warning[] warnings = null;
     while (true) {
       try {
-        return executeCallNode.executeCall(frame, function, callerInfo, state, arguments);
+        Object result = executeCallNode.executeCall(frame, function, callerInfo, state, arguments);
+        if (warnings != null) {
+          return WithWarnings.appendTo(EnsoContext.get(this), result, warnings);
+        } else {
+          return result;
+        }
       } catch (TailCallException e) {
         function = e.getFunction();
         callerInfo = e.getCallerInfo();
         arguments = e.getArguments();
+        if (warnings == null) {
+          warnings = e.getWarnings();
+        }
```

Comment:
What if `warnings != null`?

Shouldn't we somehow concatenate the existing warnings with the new ones?